src/braket/default_simulator/linalg_utils.py

```diff
@@ -17,7 +17,47 @@
 import numpy as np
 
 
-def multiply_matrix(state: np.ndarray, matrix: np.ndarray, targets: Tuple[int, ...]) -> np.ndarray:
+def apply_operation(
```

Comment:
Since this has the same signature as `multiply_matrix`, I'd make this new function `multiply_matrix` and the old one a private `_multiply_matrix`.

Reply:
It's a slightly different signature, as it takes into account control qubits, while `multiply_matrix` only does the matrix multiplication

Reply:
The way I see it, this still fits into `multiple_matrix` because `multiply_matrix` already goes a bit beyond pure matrix multiplication by specifying a target, and not specifying `controls` or `control_state` will do exactly what users expect.

As well the name `apply_operation`
1. Goes a bit beyond just just linear algebra, and
2. Could have a meaning beyond just multiplying a matrix into a state; for example, applying a noise operation applies a sum of products.